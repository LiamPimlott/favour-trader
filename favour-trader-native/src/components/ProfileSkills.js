import React, { Component } from 'react';
import { View, StyleSheet, Text, FlatList } from 'react-native';
import { List, ListItem, ButtonGroup } from 'react-native-elements'; // 0.17.0

import "@expo/vector-icons"; // 5.2.0

export default class ProfileSkills extends Component {
  state = {
    index: 0
  }

  updateIndex = (index) => {
    this.setState({ index })
  }
  renderSeparator = () => {
    return (
      <View
        style={{
          height: 1,
          width: "86%",
          backgroundColor: "#CED0CE",
          marginLeft: "14%"
        }}
      />
    );
  };

  renderList() {
    const { has } = this.props;
    const { wants } = this.props;
    const list = [
      {
        title: 'Appointments',
        icon: 'av-timer'
      },
      {
        title: 'Trips',
        icon: 'flight-takeoff'
      },
    ]
    if (this.props.has && this.props.wants) {
      if (this.state.index == 0) {
        return (
          has.map((item, i) => (
            <ListItem
              hideChevron={true}
              key={i}
              title={item.category.skill}
            />
          ))
        );
      }
      if (this.state.index == 1) {
        return (
          wants.map((item, i) => (
            <ListItem
              hideChevron={true}
              key={i}
              title={item.category.skill}
            />
          ))
        );

      }
    }
  }

  render() {
    return (
      <View >
        <ButtonGroup
          selectedBackgroundColor="pink"
          onPress={this.updateIndex}
          selectedIndex={this.state.index}
          buttons={['Has', 'Wants']}
          containerStyle={{ height: 30 }} />
        <View >
          <List >
            {this.renderList()}
          </List>
        </View>
      </View>

    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 20,
    justifyContent: 'center',
  },

});