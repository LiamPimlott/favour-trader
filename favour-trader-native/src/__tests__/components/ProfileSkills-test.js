import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import moxios from "moxios";
import ProfileSkills from '../../components/ProfileSkills.js';
import toJson from 'enzyme-to-json'; //added this line

beforeEach(function () {
    const Adapter = require("enzyme-adapter-react-16");
    configure({adapter: new Adapter()});
    moxios.install();
});

describe('Testing ProfileSkills component', () => {
    it('renders as expected', () => {
        const wrapper = shallow(
            <ProfileSkills />
        );
        expect(toJson(wrapper)).toMatchSnapshot(); //edited this line
    });
});