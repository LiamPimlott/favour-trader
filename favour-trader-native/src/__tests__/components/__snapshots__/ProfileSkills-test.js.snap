// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ProfileSkills component renders as expected 1`] = `
<View>
  <Card>
    <ButtonGroup
      buttons={
        Array [
          "Has",
          "Wants",
        ]
      }
      containerBorderRadius={3}
      containerStyle={
        Object {
          "height": 30,
        }
      }
      onPress={[Function]}
      selectMultiple={false}
      selectedBackgroundColor="pink"
      selectedIndex={0}
      selectedIndexes={Array []}
    />
    <View>
      <List />
    </View>
  </Card>
  <View />
</View>
`;
